packages/gitgraph-core/src/graph-rows/regular.ts

```diff
@@ -14,7 +16,10 @@ export class RegularGraphRows<TNode> {
   }
 
   public getMaxRow(): number {
-    return uniq(Array.from(this.rows.values())).length - 1;
+    if (this.maxRowCache === undefined) {
+      this.maxRowCache = uniq(Array.from(this.rows.values())).length - 1;
+    }
+    return this.maxRowCache;
```

Comment:
👍 Thanks for adding cache.

The only thing is that we should invalidate the cache whenever `this.rows` changes. That means we should reset it to `undefined` in the `computeRowsFromCommits()` method. It's fine with current usage as we probably only call the compute method once per instantiation, but it would prevent future caching issues.

I'll do that after merge.